universal-application-tool-0.0.1/app/services/applicant/ReadOnlyApplicantProgramServiceImpl.java

```diff
@@ -28,6 +36,10 @@ protected ReadOnlyApplicantProgramServiceImpl(
     ImmutableList<Block> blocks =
         programDefinition.blockDefinitions().stream()
             .map(blockDefinition -> new Block(blockDefinition.id(), blockDefinition, applicantData))
+            .filter(
+                block ->
+                    !block.isCompleteWithoutErrors()
+                        || block.wasCompletedInProgram(programDefinition.id()))
```

Comment:
Question (blocking for an answer): if a block is completed, but only partially in this program (some of the questions are answered in other programs), it is not included here. Shall we include them as well?

Reply:
Good catch! Bion caught this too - I'll update `wasCompletedInProgram` to check for any questions in this block that were completed in this program, instead of all questions